controllers/external-accounts.js

```diff
@@ -62,6 +63,22 @@ const syncExternalAccountData = async (req, res) => {
       };
     });
 
+    rdsUserData.forEach((rdsUser) => {
+      if (
+        rdsUser.roles?.in_discord === true &&
+        !discordUserData.some((discordUser) => discordUser.user.id === rdsUser.discordId)
```

Comment:
we're already looping over discord user data, I think we can leverage the `discordUserData.forEach()` loop and remove this loop entierely

Reply:
Done